### PR TITLE
V65/processors

### LINKED ISSUE #3506: Ingest Pipeline processors if condition support
**NEST/Elasticsearch.Net version**:
6.3.1

**Elasticsearch version**:
6.5.0

**Description of the problem including expected versus actual behavior**:
All [processors](https://www.elastic.co/guide/en/elasticsearch/reference/6.5/ingest-processors.html) have an `if` condition option, but this is not exposed in for example the `AppendProcessorDescriptor` (or any other `IProcessor`).

**Steps to reproduce**:
 1. Create pipeline:
```
await _service.PutPipelineAsync("my-pipeline", descriptor => descriptor
    .Description("My Pipeline")
    .Processors(processorsDescriptor => processorsDescriptor
        .Append<object>(appendDescriptor => appendDescriptor
            .Field("my_field")
            .Value("the_value")
        )
    )
);
```
 2. Note there's no option for `.If()` like there is for `.OnFailure()`

**Describe the feature**:
Expose a method to configure an `If` condition just like [OnFailure](https://github.com/elastic/elasticsearch-net/blob/master/src/Nest/Ingest/Processor.cs#L12)

### PR DESCRIPTION
**Pull request by Mpdreamz**
This adds missing properties and missing processors to NEST.

Add support for the dissect ingest processor
Add support for if/tag and ignore_failure on all processors fixes #3506
Add support for the drop processor
Add support for the set_security_user processor
Add support for the pipeline processor

### CODE REVIEW

File: src/Nest/Ingest/Processor.cs [hunk 1]
```diff
@@ -4,21 +4,56 @@
 
 namespace Nest
 {
+	/// <summary> Ingest pipelines are composed of one or more processors </summary>
 	public interface IProcessor
 	{
+		// TODO: eventhough this property is ignored it has a JsonProperty because our GetCachedProperties helper prefers
```
Comment (codebrain) [inline src/Nest/Ingest/Processor.cs]: TODO? (even though, needs a space)

File: src/Nest/Ingest/Processor.cs [hunk 2]
```diff
@@ -4,21 +4,56 @@
...
 		IEnumerable<IProcessor> OnFailure { get; set; }
+
+		/// <summary> A painless script predicate that can control whether this processor should be executed or not </summary>
+		[JsonProperty("if")]
+		string If { get; set; }
+
+		/// <summary>
+		/// A tag is simply a string identifier of the specific instantiation of a certain processor
+		/// in a pipeline. The tag field does not affect the processor’s behavior, but is very useful
+		/// for bookkeeping and tracing errors to specific processors./
```
Comment (codebrain) [inline src/Nest/Ingest/Processor.cs]: Trailing /

File: src/Nest/Ingest/Processors/DissectProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	/// <summary>
+	/// Similar to the Grok Processor, dissect also extracts structured fields out of a single text field
+	/// within a document. However unlike the Grok Processor, dissect does not use Regular Expressions.
+	/// This allows dissect’s syntax to be simple and for some cases faster than the Grok Processor.
```
Comment (codebrain) [inline src/Nest/Ingest/Processors/DissectProcessor.cs]: This allows dissect’s syntax to be simple and, for some cases, faster than the Grok Processor.

File: src/Nest/Ingest/Processors/PipelineProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	/// <summary> Executes another pipeline.</summary>
+	[JsonObject(MemberSerialization.OptIn)]
+	[JsonConverter(typeof(ProcessorJsonConverter<PipelineProcessor>))]
+	public interface IPipelineProcessor : IProcessor
+	{
+		//TODO this property clashes with the Name property on the IProcessor, need to rename base in master
```
Comment (codebrain) [inline src/Nest/Ingest/Processors/PipelineProcessor.cs]: TODO?

Comment (Mpdreamz) [reply src/Nest/Ingest/Processors/PipelineProcessor.cs]: We can't fix this in `6.x` this needs to be changed in `master` updated the TODO to mention this. 👍

**Comment by Mpdreamz** [comment]
:heart: the nitpicks thanks @codebrain 👍

**Pull request merged**